xtable-core/src/test/java/org/apache/xtable/parquet/ITParquetConversionSource.java

```
@@ -0,0 +1,566 @@
+            RowFactory.create(
+                4, "David", false, 29.5, new Timestamp(System.currentTimeMillis() + 3000)),
+            RowFactory.create(
+                5, "Eve", true, 22.2, new Timestamp(System.currentTimeMillis() + 4000)));
+
+    schema =
+        DataTypes.createStructType(
+            new StructField[] {
+              DataTypes.createStructField("id", DataTypes.IntegerType, false),
+              DataTypes.createStructField("name", DataTypes.StringType, false),
+              DataTypes.createStructField("hasSiblings", DataTypes.BooleanType, false),
+              DataTypes.createStructField("age", DataTypes.DoubleType, false),
+              DataTypes.createStructField(
+                  "timestamp",
+                  DataTypes.TimestampType,
+                  false,
+                  new MetadataBuilder().putString("precision", "millis").build())
+            });
+    Dataset<Row> df = sparkSession.createDataFrame(data, schema);
+    String dataPath = tempDir.toAbsolutePath().toString() + "/non_partitioned_data";
+    df.write().mode(SaveMode.Overwrite).parquet(dataPath);
+    GenericTable table;
+    table =
+        GenericTable.getInstance(
+            tableName, Paths.get(dataPath), sparkSession, jsc, sourceTableFormat, false);
+    try (GenericTable tableToClose = table) {
+      ConversionConfig conversionConfig =
+          getTableSyncConfig(
+              sourceTableFormat,
+              SyncMode.FULL,
+              tableName,
+              table,
+              targetTableFormats,
+              xTablePartitionConfig,
+              null);
+      ConversionController conversionController =
+          new ConversionController(jsc.hadoopConfiguration());
+      conversionController.sync(conversionConfig, conversionSourceProvider);
+      checkDatasetEquivalenceWithFilter(
+          sourceTableFormat, tableToClose, targetTableFormats, filter);
+    } catch (URISyntaxException e) {
+      throw e;
```

No need to catch this, you can just let it throw

---

ok